.github/workflows/windows.yml

```diff
@@ -0,0 +1,143 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+---
+name: "Windows Continuous Integration"
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+      - ci
+jobs:
+  test:
+    runs-on: ubuntu-20.04
+    env:
+      CHEF_GCP_ZONE: us-east1-b
+    strategy:
+      fail-fast: false
+      matrix:
+        agent_type:
+          - ops-agent
+        version:
+          - "2.0.1"
+          - latest
+        # distro is the platform name in kitchen.yml
+        distro:
+          - windows-2019
+          - windows-2019-core
+          - windows-2016
+          - windows-2016-core
+          - windows-2012r2
+          - windows-2012r2-core
+          - windows-20h2-core
+          - windows-2004-core
+
+        exclude:
+          # invalid combos
+          - agent_type: logging
+            version: "2.0.1"
+          - agent_type: monitoring
```

Comment:
These are exclusions, which is how the CI knows to ignore them. I'm not sure if @jsirianni could do it differently. Probably best if he answers, since you're asking for combining and parameterizing these files anyway.

Reply:
The `exclude` section below can be removed, right?

```
        exclude:
          # invalid combos
          - agent_type: logging
            version: "2.0.1"
          - agent_type: monitoring
             version: "2.0.1"
```

because in line 30 we only have

```
      matrix:
        agent_type:
          - ops-agent
```

The `logging` or `monitoring` values are not in the overall `agent_type` in `windows.yml`.